types/coin.go

```diff
@@ -576,8 +573,8 @@ func (coins Coins) Sort() Coins {
 // Parsing
 
 var (
-	// Denominations can be 3 ~ 64 characters long.
-	reDnmString = `[a-z][a-z0-9/]{2,63}`
+	// Denominations can be 3 ~ 128 characters long.
+	reDnmString = `[a-z][a-zA-Z0-9/]{2,127}`
```

Comment:
yeah, I had to make this change because the string format for the hex bytes is capitalized. 

> maybe note that denominations cannot begin with a capital letter? I didn't notice this on first glance

Leaving the first letter as lowercase is ok as the expected denom would start with `ibc/<HASH>`. My personal preference to allow all caps as you could then have standard denom symbols/tickers (i.e BTC, ETH, etc). Thoughts @clevinson @alexanderbez @aaronc

Reply:
Let's make it simple and allow caps at the start. Otherwise looks good. Although I wonder if we should consider unicode. But maybe that's a separate PR.